USING_ELEKTRA.md

```diff
@@ -0,0 +1,136 @@
+ has to be of type string. Elektra will prevent you to set invalid values which will help reduce errors and misconfiguration.
+  All these data and much more (eg. default values, descriptions, etc.) are saved as 
+ [metadata](https://www.libelektra.org/devdocu/metadata)
+ for each and every key under the `spec` namespace.
+ 
+ The following sections will teach you on how to inspect all relevant settings
+  and specifications and change desired configuration settings accordingly.
+ There best way to change settings is done via `kdb set`
+ since it integrates the best with specifications for configurations.
+ 
+### Changing Configurations
+ 
+Using the native command line is the best possibility to find out which configuration settings are available.
+If you want to see all available configuration settings for the `server`, you can simply call
+```sh
+kdb ls '/sw/lcdproc/lcdd/#0/current/server'
+#> /sw/lcdproc/lcdd/#0/current/server/autorotate
+#> /sw/lcdproc/lcdd/#0/current/server/backlight
+#> /sw/lcdproc/lcdd/#0/current/server/bind
+#> /sw/lcdproc/lcdd/#0/current/server/driver
+...
+```
+To see all available configuration settings at all you can execute `kdb ls '/sw/lcdproc/lcdd/#0/current'`.
+To see the current value you can simply call `kdb get` on the respective key.
+Now lets take the curses driver for example again and lets see what is the 
+current background color is:
+```sh
+kdb get '/sw/lcdproc/lcdd/#0/current/curses/background'
+#> cyan
+```
+The result comes from the set `default` metadata which is saved for the key. 
+You can see all metadata by calling `kdb lsmeta`:
+```sh
+kdb lsmeta '/sw/lcdproc/lcdd/#0/current/curses/background'
+#> check/enum/#0
+#> check/enum/#1
+#> check/enum/#2
+#> check/enum/#3
+#> check/enum/#4
+#> check/enum/#5
+#> check/enum/#6
+#> check/enum/#7
+#> check/type
+#> default
+#> description
+#> type
+```
+Here we can als see `check/enum/#0-7` which is used by Elektra to assert
+for correct configuration settings. You can query for the values by calling
+`kdb getmeta`
+```sh
+kdb getmeta '/sw/lcdproc/lcdd/#0/current/curses/background' 'check/enum/#0'
+#> red
+```
+This `.../#<number>` notation is the typical array notation which is used by Elektra.
+If you want to query all metadata and see their values you can use this little script:
+```sh
+kdb lsmeta '/sw/lcdproc/lcdd/#0/current/curses/background' \
+| xargs -I% -n1 sh -c 'printf "% = " && kdb getmeta \
+"/sw/lcdproc/lcdd/#0/current/curses/background" "%"'
+#> check/enum/#0 = red
+#> check/enum/#1 = black
+#> check/enum/#2 = green
+#> check/enum/#3 = yellow
+#> check/enum/#4 = blue
+#> check/enum/#5 = magenta
+#> check/enum/#6 = cyan
+#> check/enum/#7 = white
+#> check/type = enum
+#> default = cyan
+#> description = background color when "backlight" is off
+#> type = enum
+```
+In future versions we might include a command which gives you all metadata 
```

Comment:
We should do it now.

Reply:
I agree but until it is implemented, how should I deal with it in the documentation?

Reply:
Implementing it, it is the easiest way.